packages/sync/src/modules/Terms.php

```diff
@@ -275,4 +285,22 @@ public function expand_term_taxonomy_id( $args ) {
 			'previous_end' => $previous_end,
 		);
 	}
+
+	/**
+	 * Appends a term object to a row object from the term_relationships table.
+	 * This aids WordPress.com in re-syncing term_relationship tables that are out of sync.
+	 *
+	 * @access public
+	 *
+	 * @param object $relationship A row object from the term_relationships table.
+	 * @return object A row object from the term_relationships table with corresponding term object appended.
+	 */
+	public function add_term_to_relationship( $relationship ) {
+		return (object) array(
+			'object_id'        => $relationship->object_id,
+			'term_taxonomy_id' => $relationship->term_taxonomy_id,
+			'term_order'       => $relationship->term_order,
+			'term'             => get_term_by( 'term_taxonomy_id', $relationship->term_taxonomy_id ),
```

Comment:
This seems suboptimal: the `$term` object will contain all of the above, so we'll be sending some of the data twice. We should be able to avoid that, no?

Reply:
agreed, and fixed.